Change the dummy provider to accept placement constraints that
are specific to other real providers (such as MAAS).

This is needed by code under test that creates machines with
provider-specific constraints, and doesn't know it's actually talking
to fake juju with a dummy provider.

Index: 2.1-rc1/src/github.com/juju/juju/provider/dummy/environs.go
===================================================================
--- 2.1-rc1.orig/src/github.com/juju/juju/provider/dummy/environs.go
+++ 2.1-rc1/src/github.com/juju/juju/provider/dummy/environs.go
@@ -656,9 +656,11 @@ func (e *environ) checkBroken(method str
 
 // PrecheckInstance is specified in the state.Prechecker interface.
 func (*environ) PrecheckInstance(series string, cons constraints.Value, placement string) error {
-	if placement != "" && placement != "valid" {
-		return fmt.Errorf("%s placement is invalid", placement)
-	}
+	// XXX fake-juju: start patch
+	// if placement != "" && placement != "valid" {
+	//         return fmt.Errorf("%s placement is invalid", placement)
+	// }
+	// XXX fake-juju: end patch
 	return nil
 }
 
